mozci/push.py

```diff
@@ -1030,3 +1030,98 @@ def make_push_objects(**kwargs):
             cur._child = pushes[i + 1]
 
     return pushes
+
+
+def make_summary_objects(from_date, to_date, branch, type):
+    """Returns a list of summary objects matching the parameters.
+
+    When invoked with a `type` argument, this method will return a list of
+    Summary objects of the corresponding type. For example, `type='group'` will
+    return a list of GroupSummary objects.
+
+    Args:
+        from_date (str): String representing an acceptable date value in ActiveData.
+        to_date (str): String representing an acceptable date value in ActiveData.
+        branch (str): String that references one of the Mozilla CI's repositories.
+        type (str): String that references one of the supported Summary types.
+
+    Returns:
+        list: List of Summary objects, or an empty list.
+
+    """
+    # Retrieve the function by name using the provided `type` argument.
+    func = "__make_{}_summary_objects".format(type).lower()
+
+    # If the method name `func` (maps to either of the private methods)
+    # is not found, then return an empty list.
+    if not func:
+        return []
+
+    # Obtain list of all pushes for the specified branch, from_date and to_date.
+    pushes = make_push_objects(from_date=from_date, to_date=to_date, branch=branch)
+
+    summaries = globals()[func](pushes)
+
+    # Sort by either the label (LabelSummary) or name (GroupSummary).
+    summaries.sort(key=lambda x: (getattr(x, "label", "name")))
+    return summaries
+
+
+def __make_label_summary_objects(pushes):
```

Comment:
@ahal I've decided to simply add `__` in front to denote private method and keep these operations in this file.

I kept the `make_summary_objects` functions here because:
  - they operate on push(es)
  - `tasks.py` in my mind is scoped narrower than `push.py`

With that said, if you think it is better suited in `tasks.py`, I'm open to moving the entire set of functions that I've added in this patch there. I see them living as top-level functions in the file.

Reply:
Yeah, that's what I had in mind (though usually a single underscore is the standard, double underscores typically denote magic methods on classes)

Reply:
I agree having it in `tasks.py` makes sense as well.